src/OnchainKitProvider.tsx

```diff
@@ -48,6 +56,32 @@ export function OnchainKitProvider({
     return onchainKitConfig;
   }, [address, apiKey, chain, config, projectId, rpcUrl, schemaId]);
 
+  // Check the context for WagmiProvider
+  // Wagmi configuration defaults to the provided config if it exists
+  // Otherwise, use the OnchainKit-provided Wagmi configuration
+  let providedConfig = null;
+  try {
+    providedConfig = useConfig();
+  } catch (error) {
+    if (!(error instanceof WagmiProviderNotFoundError)) {
+      console.error('Error fetching config, using OnchainKit defaults:', error);
+      throw error;
+    }
+  }
+
+  // If WagmiProvider is not found, return the context with defaulted parent providers
+  if (!providedConfig) {
+    return (
+      <WagmiProvider config={getDefaultConfig({ apiKey, config })}>
+        <QueryClientProvider client={new QueryClient()}>
```

Comment:
`create wagmi` wraps these in a `useState` but not sure the benefits of doing so

```
  const [config] = useState(() => getConfig())
  const [queryClient] = useState(() => new QueryClient())
```

Reply:
should the Wagmi configuration (and/or the query client) be memo-ized?

Reply:
```
  const defaultConfig = useMemo(() => getDefaultConfig({ apiKey, config }), [apiKey, config]);
  const queryClient = useMemo(() => new QueryClient(), []);

  if (!providedConfig) {
    return (
      <WagmiProvider config={defaultConfig}>
        <QueryClientProvider client={queryClient}>
          <OnchainKitContext.Provider value={value}>
            {children}
          </OnchainKitContext.Provider>
        </QueryClientProvider>
      </WagmiProvider>
    );
  }
```

if we do this, are these values created even if they aren't used? not sure on the client-side implications, since if a parent `WagmiProvider` is provided these would be essentially duplicates and take up storage/memory

Reply:
you could split this default provider out to another component to solve that